packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesRemoteAppLoader.m

```diff
@@ -32,9 +35,27 @@ - (void)loadUpdateFromUrl:(NSURL *)url
                     error:(EXUpdatesAppLoaderErrorBlock)error
 {
   self.manifestBlock = manifestBlock;
-  self.successBlock = success;
   self.errorBlock = error;
+
+  UM_WEAKIFY(self)
+  self.successBlock = ^(EXUpdatesUpdate * _Nullable update) {
+    UM_STRONGIFY(self)
+    // even if update is nil (meaning we didn't load a new update),
+    // we want to persist the header data from _remoteUpdate
+    if (self->_remoteUpdate) {
+      dispatch_async(self.database.databaseQueue, ^{
+        NSError *serverDataError;
+        [self.database setServerDataWithManifest:self->_remoteUpdate error:&serverDataError];
+        if (serverDataError) {
+          NSLog(@"Error persisting header data to disk: %@", serverDataError.localizedDescription);
+        }
+      });
+    }
+    success(update);
```

Comment:
good catch. I don't think we can get into a bad/inconsistent state in terms of loading, because the database queue is serial. But, we could potentially report that an update finished loading successfully, but if there was an error saving the metadata then the new update might not load, which could be confusing. Will follow up in a separate PR.